app/src/main/java/com/kickstarter/ui/activities/MessageThreadsActivity.java

```diff
@@ -81,4 +86,35 @@ protected void onDestroy() {
     this.recyclerViewPaginator.stop();
     this.recyclerView.setAdapter(null);
   }
+
+  // todo: make inputs and outputs
+  private void setOffsetChangedListener() {
+    this.appBarLayout.addOnOffsetChangedListener(
+      (appBarLayout, offset) -> {
+        if (offset == 0) {  // expanded
+          this.expandedToolbarTitle.setVisibility(View.VISIBLE);
+          this.collapsedToolbarTitle.setVisibility(View.INVISIBLE);
+
+        } else if (Math.abs(offset) >= appBarLayout.getTotalScrollRange()) {
+          this.expandedToolbarTitle.setVisibility(View.INVISIBLE);
+          this.collapsedToolbarTitle.setVisibility(View.VISIBLE);
+        }
+      }
+    );
+  }
+
+  private void setUnreadTextViewText(final @Nullable Integer unreadCount) {
+    if (unreadCount == null) {
+      this.unreadCountTextView.setText(this.noMessagesString);
+    } else if (unreadCount == 0) {
+      this.unreadCountTextView.setText(this.noUnreadMessagesString);
+    } else {
+      final String unreadCountString = NumberUtils.format(unreadCount);
+
+      this.unreadCountTextView.setText(
+        this.ksString.format(this.unreadCountUnreadString, "unread_count", unreadCountString)
+      );
+      this.unreadCountToolbarTextView.setText(StringUtils.wrapInParentheses(unreadCountString));
+    }
+  }
```

Comment:
yesss that makes a lot more sense, and will pull more logic out of the view. will refactor thanks!